[DOC] Create initial GitHub Pages documentation

m273d15 opened the pull request:
* Created a first default layout #108 
* Created first content in order to verify whether the layout and structure
  work for more than one page

tobous commented on docs/_config.yml:
```diff
@@ -0,0 +1,17 @@
+description: >- # this means to ignore newlines until "baseurl:"
+  Write an awesome description for your new site here. You can edit this
+  line in _config.yml. It will appear in your document head meta (for
+  Google search results) and in your feed.xml site description.
```
It seems to me like the fields email and description are still set to default values.

What is the email field used for?

Add a general description of the page? Something like "Homepage of Saros, an IDE plugin which enabled users to participate in distributed pair programming sessions."?
I found a page about how google handles these description meta tags, but I still don't know what would be best to put here. You can have a look for yourself: https://support.google.com/webmasters/answer/35624?visit_id=1-636674263008325798-3147872153&rd=1
It's the last section titled "Create good meta descriptions".

tobous commented on docs/documentation/README.md:
```diff
@@ -0,0 +1,4 @@
+# Documentation
+
+This directory contains only user specific documentation.
+Development documentation is located in the directory "contribute".
```
Is it still fitting to use the term "directory" when talking about web-pages (even though they are represented through a directory structure underneath)?

m273d15 replied:
The idea was that the README.md files are files that describe the structure of the documentation and describe how the documentation can be build with jekyll. Therefore they don't start with [frontmatter](https://jekyllrb.com/docs/frontmatter/) and should not be rendered by jekyll.

m273d15 commented:
@tobous You don't have to review the whole text. I already changed content that is out-dated, but a whole rework is necessary.

I changed the description of the site and substituted the former twitter variable with the seo twitter variable.

fzieris commented:
> m273d15 requested a review from fzieris

Anything in particular? The technical infrastructure or the content?

tobous commented:
You could add the developer mailing list as the email address. But I am not really sure what it would be used for.
Looks good to me otherwise.

fzieris commented on docs/README.md:
```diff
@@ -0,0 +1,19 @@
+
+### Compile and show documentation
+* Move into this directory (`<repository_dir>/docs`) 
+* Execute `bundle exec jekyll serve`
```
Maybe it's my Ruby setup (although I followed the install wizard), but `bundle exec jekyll serve` won't work: "bundler: command not found: jekyll".
Calling `jeykll serve` directly works, though.

m273d15 replied:
I tried the manual again. I followed the installation manual and used a fresh docker image ubuntu:18.04. The only challenge is to execute `bundle install`. I had to install the packages `ruby ruby-dev build-essential git libpng-dev zlibc zlib1g zlib1g-dev`.
After executing 'bundle install' in the docs dir everything works fine. Did you use Windows or a Linux-VM?

fzieris commented on docs/_config.yml:
```diff
@@ -0,0 +1,13 @@
+title: Saros
+subtitle: Distributed Party Programming
```
Preferred subtitle: "Distributed Collaborative Programming"

fzieris commented on docs/_layouts/default.html:
```diff
@@ -0,0 +1,70 @@
+      </footer>
+    </div>
+
+    {% if site.google_analytics %}
```
Please remove Google Analytics.

fzieris commented on docs/_layouts/default.html:
```diff
@@ -0,0 +1,70 @@
+    <div id="footer_wrap" class="outer">
+      <footer class="inner">
+        <div id="copyright">Copyright &copy; {{ site.time | date: '%Y' }}, <a href="{{ '/documentation/history.html' | relative_url }}">Saros Team</a></div>
+        <a href="http://www.mi.fu-berlin.de/fb/impressum/" id="impressum" target="_blank">Impressum</a>
```
"Imprint"

fzieris commented on docs/index.md:
```diff
@@ -0,0 +1,38 @@
+Programmers](images/comics/dpp_0.png){:class="center"}
+
+
+## What others think 
```
Preferred order:

* Brief explanation of Saros
* "Get Saros"
* "Good Starting Points"
* "What others think"

fzieris commented on docs/_config.yml:
```diff
@@ -0,0 +1,13 @@
+title: Saros
+subtitle: Distributed Party Programming
```
`jekyll serve` wouldn't serve the site directly:

```
YOUR SITE COULD NOT BE BUILT:
------------------------------------
No repo name found. Specify using PAGES_REPO_NWO environment variables,
'repository' in your configuration, or set up an 'origin' git remote pointing
to your github.com repository.
```

It worked after adding:

```diff
+ repository: saros-project/saros
```

m273d15 replied:
In my test, the command works after installing git. I assumed that git is available. Does this error occur with a git installation?

m273d15 replied:
I will add the repository information in order to avoid this error

fzieris commented on docs/documentation/index.md:
```diff
@@ -0,0 +1,109 @@
+
+  Two or more participants work
+  together in a loosely coupled
+  fashion.** They **work independently
```
Formatting is messed up.

fzieris commented on docs/documentation/getting-started.md:
```diff
@@ -0,0 +1,280 @@
+
+1.  You can **create a new account** within Saros by using the
+    FU-Berlin servers.
+2.  You can **use an existing account.**\
```
Those backslashes are displayed as-is. I saw this in multiple places.

fzieris commented on docs/documentation/getting-started.md:
```diff
@@ -0,0 +1,280 @@
+![Share
+Projects](images/3_GettingStartedShareProjects_0.png)
+
+#### a) Start a session ([host](GettingStarted.md#Step%200:%20The%20Host))
```
Dead link

fzieris commented on docs/documentation/getting-started.md:
```diff
@@ -0,0 +1,280 @@
+session:**
+
+Click here to leave the current session. If you are the
+[host](GettingStarted.md#Step%200:%20The%20Host), the session will be
```
Dead link

fzieris commented on docs/documentation/getting-started.md:
```diff
@@ -0,0 +1,280 @@
+doing:
+
+-   In the package explorer (or resource navigator):
+    -   ![active file](images/active_file.png) A
```
Unlike the other icons above, these two icons have a prominent border. 
![image 5](https://user-images.githubusercontent.com/4941859/42878689-37e8690a-8a8e-11e8-8363-f594d003bd96.png)

fzieris commented on docs/documentation/compatibility.md:
```diff
@@ -0,0 +1,53 @@
+---
```
Badly out of date, no need to migrate this to GitHub pages.

fzieris commented on docs/documentation/faq.md:
```diff
@@ -0,0 +1,149 @@
+### Is Saros language dependent or which languages does Saros support?
+
+No, Saros is not language-dependent as it works on a character-level.
+For a list of programming language plugins explicitly tested see:
```
Remove all but the first sentence of this answer.

fzieris commented on docs/downloads/index.md:
```diff
@@ -0,0 +1,16 @@
+
+# {{ page.title }}
+
+See the [installation documentation](/documentation/installation.md) for more information about prerequisites.
```
Dead link

fzieris commented on docs/contribute/guidelines.md:
```diff
@@ -0,0 +1,269 @@
+    line to make it easier for people to understand what your commit was
+    about:
+
+* `[NOP]` - This commit did not have any effect and only concerns whitespace, removing unused methods, fixing documentation typos, etc.
```
Needs to be a table

m273d15 replied:
I will do this, but I wil also add a TODO, because this will lead to a less readable table, because the default width of the `main_content` div is to small.

m273d15 replied:
The TODO is not necessary the table looks good. It seems like as if my styling or formatting changes solved the problem.

fzieris commented on docs/contribute/processes/review.md:
```diff
@@ -0,0 +1,248 @@
+#### b) Format of your pull request
+
+*   Write a **concise commit message** following our
+    [guidelines](coderules.1.md#before_committing). If you have
```
Dead link

fzieris commented on docs/_data/contribute/sidebar.yml:
```diff
@@ -0,0 +1,11 @@
+- title: Review Process
+  url: /contribute/processes/review.html
+
+- title: Development Process
```
"Development Process" should be the first entry

fzieris commented on docs/contribute/processes/development.md:
```diff
@@ -0,0 +1,16 @@
+
+# {{ page.title }}
+
+Our development workflow bases on the standard [GitHub workflow](https://guides.github.com/introduction/flow/) and consists of the following steps:
```
Provide a link to the review process

fzieris commented on docs/contribute/guidelines.md:
```diff
@@ -0,0 +1,269 @@
+---
+title: Coding and Commit Guidelines
```
This list looks _much_ shorter than the original one. Which sections did you remove and why?

m273d15 replied:
This change does not contain a full documentation. Therefore the guideline page does not contain all information.

I had the following plan with the [old content](http://www.saros-project.org/guidelines):
* Ticketing - is out-dated and will be removed
* Contribution Rules- referenced in #111 (see file old/contribution.md)
* Coding Conventions - referenced in #111 (see file old/coderules.1.md)
* Usability - referenced in #111 (see file old/usability.md)
* Continuous Integration with Jenkins - out-date and will be replaced

The files references in #111 will be transferred and reworked.

fzieris commented on docs/_config.yml:
```diff
@@ -1,9 +1,10 @@
 title: Saros
-subtitle: Distributed Party Programming
+subtitle: Distributed Collaborative Progamming
```
"Prog**r**amming"

m273d15 commented on docs/index.md:
```diff
@@ -5,6 +5,22 @@ title: Home
 ![Comic: Distributed Pair
 Programmers](images/comics/dpp_0.png){:class="center"}
 
+<!-- TODO: add short description of saros-->
```
Added a TODO, because I am not sure how we want to design the first page.

If you take a look at other dev tools ([gradle](https://gradle.org/), [docker](https://www.docker.com/), [npm](https://www.npmjs.com/), [intellij](https://www.jetbrains.com/idea/), [eclipse](https://www.eclipse.org/ide/)), the index page is only for marketing.
  * very short description of the tool
  * highlight main feature
  * cute images

Probably we want an index page with another design than the other pages.

m273d15 commented:
@fzieris Do you still request changes?

m273d15 commented:
@fzieris Could you please rereview the latest change. I tested this change in a forked repository and I had to fix the way the sidebar links are created. Otherwise the links work only if you render the pages locally, but not in GitHub.
